tests/test_box_yaml_sequence.py

```diff
@@ -13,6 +14,20 @@ def test_simple_sequence():
     # didn't make any edits, this should be fine
     assert stream.text == b"- foo\n- bar\n"
 
+def test_edit_sequence():
+    stream = parse_str("- foo\n- bar\n")
+    stream.append(PyScalarString("baz", QuoteStyle.BARE))
```

Comment:
Not the final interface -- this is a little verbose until I get style replication in a subsequent PR.